src/system-update-generator/system-update-generator.c

```diff
@@ -18,21 +19,22 @@
 static const char *arg_dest = NULL;
 
 static int generate_symlink(void) {
-        const char *p = NULL;
+        FOREACH_STRING(p, "/system-update", "/etc/system-update") {
+                if (laccess(p, F_OK) < 0) {
+                        if (errno == ENOENT)
+                               continue;
 
-        if (laccess("/system-update", F_OK) < 0) {
-                if (errno == ENOENT)
-                        return 0;
+                        return log_error_errno(errno, "Failed to check for system update: %m");
+                }
 
-                log_error_errno(errno, "Failed to check for system update: %m");
-                return -EINVAL;
-        }
+                p = strjoina(arg_dest, "/" SPECIAL_DEFAULT_TARGET);
```

Comment:
strictly speaking you are using strjoina() (and hence stack allocation via alloca() in a loop here). Even though you immediately leave the loop afterwards this is still icky, and might confuse static checkers and such. Hence, please rework this to use dynamic memory, and in fact `path_join()` + `_cleanup_free_`